spec/factories/lectures.rb

```diff
@@ -54,6 +54,21 @@
       end
     end
 
+    trait :with_title do
+      transient do
+        title { nil }
+      end
+      # also see https://github.com/thoughtbot/factory_bot/issues/1391
+      course { association :course, title: title }
+    end
+
+    trait :with_teacher_by_id do
+      transient do
+        teacher_id { nil }
+      end
+      teacher { User.find(teacher_id) }
```

Comment:
Same as above: I would expect an exception for the default value. Is this intended?

Reply:
Here, I think I'd also expect an exception because the trait is called "..._by_id", so we expect the caller to pass in an id from the outside.